.github/workflows/publish_lambda_layer.yaml

```diff
@@ -0,0 +1,99 @@
+# © 2023 SolarWinds Worldwide, LLC. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at:http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+name: Publish Lambda Layer to staging
+
+on:
+  workflow_call:
+    inputs:
+      artifact-name:
+        description: 'Name of actions/upload-artifact result'
+        required: true
+        type: string
+      component-version:
+        description: 'Version of APM Python library in this release'
+        required: true
+        type: string
+      architecture:
+        description: 'Must be x86_64 or arm64'
+        required: true
+        type: string
+      runtimes:
+        description: 'Space-delimited list of compatible runtimes'
+        required: true
+        type: string
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  publish_layer:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        aws_region: 
+          - ap-northeast-1
+          - ap-northeast-2
+          - ap-south-1
+          - ap-southeast-1
+          - ap-southeast-2
+          - ca-central-1
+          - eu-central-1
+          - eu-north-1
+          - eu-west-1
+          - eu-west-2
+          - eu-west-3
+          - sa-east-1
+          - us-east-1
+          - us-east-2
+          - us-west-1
+          - us-west-2
+
+    steps:
+      - name: Construct Layer Name
+        shell: bash
+        run: |
+          LAYER_NAME=solarwinds-apm-python
+          
+          if [[ "${{ inputs.architecture }}" != "x86_64" ]] && [[ "${{ inputs.architecture }}" != "arm64" ]] ; then
+            echo "ERROR: architecture must be one of x86_64, arm64"
+            exit 1
+          else
+            LAYER_NAME=$LAYER_NAME-${{ inputs.architecture }}
+            echo "ARCH=${{ inputs.architecture }}" >> $GITHUB_ENV
+          fi
+          
+          COMPONENT_VERSION_UNDERSCORES=$(echo ${{ inputs.component-version }} | sed -r 's/\./_/g')
+          LAYER_NAME=$LAYER_NAME-$COMPONENT_VERSION_UNDERSCORES
+          echo "LAYER_NAME=$LAYER_NAME" >> $GITHUB_ENV
+          
+          echo GITHUB_ENV:
+          cat $GITHUB_ENV
+
+      - name: Download built layer
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ inputs.artifact-name }}
```

Comment:
In each of the build-for-<architecture> workflows, the artifact is uploaded with the same name `solarwinds_apm_lambda.zip`.  I'm wondering if in this publish step, we're guaranteed to be using the correct one, i.e. would the aarch64 artifact have replaced the one built for x86 while we're publishing for x86?

From quick read of about [artifacts](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts) it seems this artifact might be shared since these build-for-<architecture> workflows are part of the main "caller" workflow.  It may be safer to just name the artifacts with architecture appended?

Reply:
Good call! Changed in workflows and lambda target in https://github.com/solarwinds/apm-python/pull/221/commits/aa42a89813cca8f258b86c613a3b417bed72553b